### PR TITLE
Download cv2pdb, generate zip file with debug symbols for winagent

### PR DESCRIPTION
**Pull request by HanesSciarrone**
|Related issue|
|---|
|#1332|

## Description

Hi team

This PR aims to build the zip with the debug symbols. The file should be built when we want to create a Windows package. After some testing, I was able to reach the following results.

- Generation and compression of debug symbols in a zip file
 
image

- Output then ran the script `sudo ./generate_compiled_windows_agent.sh -b dev-debug-symbols -c /home/hanes/wazuh-tools/utils/cv2pdb`

image

- Debug symbols compressed

image

Hanes

- Build the package in any supported platform
  - [ ] Linux
  - [x] Windows
  - [ ] macOS
  - [ ] Solaris
  - [ ] AIX
  - [ ] HP-UX
- [x] Package installation
- [ ] Package upgrade
- [ ] Package downgrade
- [ ] Package remove
- [ ] Package install/remove/install
- [ ] Change added to CHANGELOG.md

- Tests for Linux RPM
  - [ ] Build the package for x86_64
  - [ ] Build the package for i386
  - [ ] Build the package for armhf
  - [ ] Build the package for aarch64
  - [ ] `%files` section is correctly updated if necessary
- Tests for Linux deb
  - [ ] Build the package for x86_64
  - [ ] Build the package for i386
  - [ ] Build the package for armhf
  - [ ] Build the package for aarch64
  - [ ] Package install/remove/install
  - [ ] Package install/purge/install
  - [ ] Check file permissions after installing the package
- Tests for macOS
  - [ ] Test the package from macOS Sierra to Mojave
- Tests for Solaris
  - [ ] Test the package on Solaris 10
  - [ ] Test the package on Solaris 11
  - [ ] Check file permissions on Solaris 11 template
- Tests for IBM AIX
  - [ ] `%files` section is correctly updated if necessary
  - [ ] Check the changes from IBM AIX 5 to 7

### CODE REVIEW

File: windows/generate_compiled_windows_agent.sh [hunk 1]
```diff
@@ -57,6 +63,14 @@ main() {
         "-h"|"--help")
             help 0
             ;;
+        "-cv"|"--cv2pdb-path")
+            if [ -n "$2" ]; then
+                DEBUG_SYMBOLS_PATH="$2"
```
Comment (jnasselle) [inline windows/generate_compiled_windows_agent.sh:68]: `DEBUG_SYMBOLS_PATH` variable name not seems to be a tools path. Output path should still be the same of the binary package
Could you please change it to `CV2PDB_TOOLS_PATH` or something similar?

File: windows/entrypoint.sh [hunk 2]
```diff
@@ -6,6 +6,7 @@ BRANCH=$1
 JOBS=$2
 DEBUG=$3
 REVISION=$4
+DEBUG_SYMBOLS_PATH=$5
```
Comment (jnasselle) [inline windows/entrypoint.sh:9]: There's no need to use another parameter.

File: windows/entrypoint.sh [hunk 3]
```diff
@@ -15,16 +16,32 @@ wget -O wazuh.zip ${URL_REPO} && unzip wazuh.zip
...
 if [[ "${DEBUG}" = "yes" ]]; then
     FLAGS+="-d "
 fi
 
+rm /wazuh-*/src/symbols/.gitignore
+
 make -C /wazuh-*/src deps TARGET=winagent ${FLAGS}
-make -C /wazuh-*/src TARGET=winagent ${FLAGS}
+
+if [ ! -z "${DEBUG_SYMBOLS_PATH}" ]; then
```
Comment (jnasselle) [inline windows/entrypoint.sh:30]: Just checking if /tools directory exist is enough to use it

File: windows/entrypoint.sh (hunk 2 above)
Comment (RamiroRD) [reply windows/entrypoint.sh:9]: I might be missing something but if this improves readability, why not add it?

**Review by jnasselle — APPROVED** [review]

**Review by RamiroRD — APPROVED** [review]

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**